web/files/files.go

```diff
@@ -293,6 +293,75 @@ func UploadMetadataHandler(c echo.Context) error {
 	return jsonapi.Data(c, http.StatusCreated, &m, nil)
 }
 
+// FileCopyHandler handles POST requests on /files/:file-id/copy
+//
+// It is used to duplicate the given file and its metadata except for
+// relationships.
+func FileCopyHandler(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	fs := instance.VFS()
+
+	fileID := c.Param("file-id")
+	olddoc, err := instance.VFS().FileByID(fileID)
+	if err != nil {
+		return WrapVfsError(err)
+	}
+
+	err = checkPerm(c, permission.GET, nil, olddoc)
+	if err != nil {
+		return err
+	}
+
+	// TODO: move block to separate method?
+	newdoc := olddoc.Clone().(*vfs.FileDoc)
+	newdoc.DocID = ""
+	newdoc.DocRev = ""
+	newdoc.DocName = olddoc.DocName + " (copy)"
+	newdoc.Metadata.RemoveCertifiedMetadata()
+	newdoc.CozyMetadata = nil
+	newdoc.ReferencedBy = nil
+	newdoc.InternalID = ""
+	newdoc.CreatedAt = time.Now()
+	newdoc.UpdatedAt = time.Now()
```

Comment:
[nitpick] I would prefer `newdoc.UpdateAt = newdoc.CreatedAt` to have exactly the same time for both